Add nmcli support for RHEL7

robert-thompson opened the pull request:
- Add bonding configuration using the NetworkManager nmcli
  command.

StafDehat commented:
Successfully tested peer detection, primary interface configuration (ie: with gateway) and secondary (ie: without gateway). Also tested --unattend and --automated. Everything worked great. Only thing I didn't test was alternative bonding modes - only tested active-backup.  Overall though, I'm happy with this PR. :+1:

troyengel commented:
I have also tested all three modes; --automated on bond0 which configures the gateway, manual run choosing the values, and --unattend mode where I specified them on a secondary bond. All worked as expected.

One fix we could make -- when RHEL7/CentOS7 is bonding via the manual method, we can suppress the final output indicating Networking needs restarted. This is not required for nmcli ops, it's all made live when it's done right away (sweet!). The exact message I mean is...

````
# python bonding.py 
Scanning for bonding peers...
Enabling interfaces..........
Sleeping 5 seconds for switch port negotiation...
Scanning............................................................................................................Done
Interface Groups:
em1 p4p1
em2 p4p2


The following bonded interfaces are already configured:

bond0
	em1
	p4p1

Interface groups available for configuration:

1) em2 p4p2
2) em1 p4p1

Which numerical interface group from above would you like to configure? (leave blank or hit enter to perform manual entry later) 1
What is the name of the bond interface you are configuring? [bond1]: 
The bonded interface will be named: bond1

Which bonding mode do you want to use for bond1? [active-backup]: 
The bonded interface will use mode active-backup

The interfaces that will be used for bond1 will be: em2 p4p2

What IP address do you want to use for the primary IP for bond1? []: 192.168.11.4
The IP address that will be used for bond1 will be: 192.168.11.4

No Network Mask was located. What Network Mask do you want to use for bond1? [255.255.255.0]: 
The Network Mask that will be used for bond1 will be: 255.255.255.0

Current default gateway details from the running configuration:
Gateway IP:  172.24.16.1
Gateway Dev: bond0
This does not take into account configurations that have not yet been loaded into the running configuration.

Change the default gateway and gateway device on this system? [N]|y: N
The default gateway will NOT be changed for bond1

Backing up existing ifcfg files to /etc/sysconfig/network-scripts/bond1-bak-2016-01-19
Configuring bonding...

Bonding has been configured! The only thing left is to restart networking.
````

That last output is what could be suppressed for nmcli based ops in order to not give the wrong impression to the end-user.

robert-thompson commented:
Added a commit to remove `ipv4.ignore-auto-dns`. This commit can be rebased and squashed once this PR has finished review. I just wanted to preserve the history for this PR (aka that the testing was done against that specific commit id).
